### PR TITLE
Removed Army::GetUniqueCount() and replaced it with AllTroopsAreSame()

### PR DESCRIPTION
**Pull request by vincent-grosbois**
The new version doesn't produce any allocation

Also removed AllTroopsIsRace(): unused

### CODE REVIEW

File: src/fheroes2/army/army.cpp [hunk 1]
```diff
@@ -409,17 +400,21 @@ void Troops::MoveTroops( Troops & from )
...
+// Return true when all valid troops have the same ID, or when there are no troops
+bool Troops::AllTroopsAreTheSame( void ) const
 {
-    std::set<int> monsters;
-
-    for ( size_t idx = 0; idx < size(); ++idx ) {
-        const Troop * troop = operator[]( idx );
-        if ( troop && troop->isValid() )
-            monsters.insert( troop->GetID() );
+    const Troop * first_troop = nullptr;
```
Comment (ihhub) [inline src/fheroes2/army/army.cpp:406]: We could avoid pointer usage by having `int firstMonsterId = UNKNOWN;` What do you think?

File: src/fheroes2/army/army.cpp [hunk 2]
```diff
@@ -1086,15 +1081,15 @@ int Army::GetMoraleModificator( std::string * strs ) const
         ++count;
         r = Race::NONE;
     }
-    const u32 uniq_count = GetUniqueCount();
+    const bool has_different_troops = !AllTroopsAreTheSame();
```
Comment (ihhub) [inline src/fheroes2/army/army.cpp:1084]: Could we please use camelCase variable naming as per [our code style](https://github.com/ihhub/fheroes2/wiki/Code-style)?

**Review by ihhub — COMMENTED** [review]
Hi @vincent-grosbois , I left just 2 comments. Would you mind to take a look?

**Comment by vincent-grosbois** [comment]
all done!

**Comment by ihhub** [comment]
@vincent-grosbois , thank you very much for this improvement!

**Pull request merged**